### ISSUE TITLE
OS.window_size and OS.window_position properties will return useless default values if the window is minimized.

### ISSUE THREAD

**Issue by API-Beast** [godotengine/godot#20877]
**Godot version:**
v3.0.6
v3.1.dev.calinou.b293dd3

**OS/device including version:**
Windows 8.1

**Issue description:**
OS.window_size will return (0, 0)
OS.window_position will return (-32000, -32000)

Should return the actual values. Can be worked around by temporarily setting OS.window_minimized to false.

```swift
extends Node2D

func _ready():
  OS.window_minimized = true
  printt(OS.window_size, OS.window_position)
  OS.window_minimized = false
  printt(OS.window_size, OS.window_position)
```

**Comment by ivodopiviz** [comment]
Yeah, I did a quick test and using rcNormalPosition seems to be the way to get the expected results. But what about macOS and X11? How do they behave?

**Comment by akien-mga** [comment]
Seems to work fine on X11:
```gdscript
extends Node2D

func _ready():
  OS.window_minimized = true
  printt(OS.window_size, OS.window_position)   # prints (1024, 600)     (452, 269)
  OS.window_minimized = false
  printt(OS.window_size, OS.window_position)   # prints (1024, 600)     (452, 269)
```

**Comment by ordigdug** [comment]
On MacOS:

```
extends Node2D

func _ready():
    printt(OS.window_size, OS.window_position)   # prints (1024, 600)     (448, 240)
    OS.window_minimized = true
    printt(OS.window_size, OS.window_position)   # prints (1024, 600)     (448, 240)
    OS.window_minimized = false
    printt(OS.window_size, OS.window_position)   # prints (1024, 600)     (448, 240)
```

**Comment by ivodopiviz** [comment]
Guess that settles it, thanks guys. 

As I said before, the WINDOWPLACEMENT structure seems to do the trick to get the proper values. Unfortunately, it doesn't play well with the older API used in os_windows so there's a little extra work involved in fixing this.

**Comment by marcelofg55** [comment]
This should be fixed by #24078, if you can please test it again with latest master. Thanks!

**Comment by akien-mga** [comment]
Closing as fixed by #24078 so that we don't forget, but please comment to confirm that it does work as expected. If not, we can reopen the issue.
